add shape check for random_samples in fractional_max_pool{2d|3d}

mingfeima opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #89992

This PR add shape checks for `random_samples` in fractional_max_pool2d and fractional_max_pool3d.,
to provide more meaningful warnings instead of SegFault when the input is illegal.

For more details, please check https://github.com/pytorch/pytorch/issues/89648

mingfeima committed:
add shape check for random_samples in fractional_max_pool{2d|3d}

[ghstack-poisoned]

mingfeima committed:
Update on "add shape check for random_samples in fractional_max_pool{2d|3d}"


This PR add shape checks for `random_samples` in fractional_max_pool2d and fractional_max_pool3d.,
to provide more meaningful warnings instead of SegFault when the input is illegal.

For more details, please check https://github.com/pytorch/pytorch/issues/89648

[ghstack-poisoned]

jgong5 commented on aten/src/ATen/native/FractionalMaxPooling.h:
```diff
@@ -0,0 +1,80 @@
+
+  TORCH_CHECK(
+      input.scalar_type() == randomSamples.scalar_type(),
+      "Expect _random_samples to have the dtype as input");
```
```suggestion
      "Expect _random_samples to have the same dtype as input");
```

mingfeima replied:
fixed!

jgong5 reviewed (approved):
LGTM. A minor revision about an error message. Please check. Also, shall we add UT to cover this?

mingfeima committed:
Update on "add shape check for random_samples in fractional_max_pool{2d|3d}"


This PR add shape checks for `random_samples` in fractional_max_pool2d and fractional_max_pool3d.,
to provide more meaningful warnings instead of SegFault when the input is illegal.

For more details, please check https://github.com/pytorch/pytorch/issues/89648

[ghstack-poisoned]

mingfeima committed:
Update on "add shape check for random_samples in fractional_max_pool{2d|3d}"


This PR add shape checks for `random_samples` in fractional_max_pool2d and fractional_max_pool3d.,
to provide more meaningful warnings instead of SegFault when the input is illegal.

For more details, please check https://github.com/pytorch/pytorch/issues/89648

[ghstack-poisoned]

mingfeima committed:
Update on "add shape check for random_samples in fractional_max_pool{2d|3d}"


This PR add shape checks for `random_samples` in fractional_max_pool2d and fractional_max_pool3d.,
to provide more meaningful warnings instead of SegFault when the input is illegal.

For more details, please check https://github.com/pytorch/pytorch/issues/89648

[ghstack-poisoned]

mingfeima committed:
Update on "add shape check for random_samples in fractional_max_pool{2d|3d}"


This PR add shape checks for `random_samples` in fractional_max_pool2d and fractional_max_pool3d.,
to provide more meaningful warnings instead of SegFault when the input is illegal.

For more details, please check https://github.com/pytorch/pytorch/issues/89648

[ghstack-poisoned]

mingfeima commented:
@pytorchbot merge

ezyang commented:
@pytorchbot merge
